shimming-toolbox/shimmingtoolbox/coils/coil.py

```diff
@@ -217,8 +221,10 @@ def __init__(self, dim_volume, affine, constraints, orders, manufacturer="", shi
 
     def _create_coil_profile(self, dim, manufacturer=None):
         # Create spherical harmonics coil profiles
-        # f0, orders
-        mesh1, mesh2, mesh3 = generate_meshgrid(dim, self.affine)
+        # Change the affine offset so that the origin is at isocenter
+        affine_origin_iso = copy.deepcopy(self.affine)
+        affine_origin_iso[:3, 3] -= self.isocenter
```

Comment:
The coordinates are of the opposite sign?

Reply:
Isocenter is at a particular position (eg: +3mm). To create coil profiles where the new origin is at +3mm, the isocenter location is subtracted from the affine such that the new affine now has an origin at isocenter.